### ISSUE TITLE
Handy hook when using with React

### ISSUE THREAD

**Issue by ramezrafla** [mudgen/runcss#14]
For anyone using React, we created a handy-dandy hook which makes it super-easy to use runCSS in any React project

```
export const useClasses = (ref) => {
  if (!ref) ref = useRef()
  useEffect(() => {
    const parent = typeof ref == 'string' ? document.querySelector(ref) : ref.current
    if (parent) {
      if (parent.className) processClasses(parent.className)
      parent.querySelectorAll('*[class]').forEach(function(element) {
        if (element.className) processClasses(element.className)
      })
    }
  })
  return ref
}
```

Here is how you would use it

**Example 1: With a reference**
```
const component = () => {
   const root = useClasses()
   return (
       <div ref={root} className="p-10 border border-gray-200 rounded">
            <img src="..." className="rounded my-10"/>
       </div>
   )
}
```

**Example 2: With an id**
```
const component = () => {
   const root = useClasses('#my-component')
   return (
       <div id="my-component" className="p-10 border border-gray-200 rounded">
            <img src="..." className="rounded my-10"/>
       </div>
   )
}
```

**Comment by lucafabbian** [comment]
This should not be necessary anymore thanks to the new watcher API based on [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).

**Comment by ramezrafla** [comment]
Thanks @lucafabbian 

I gave the mutation observers a spin and it sporadically misses classes. We have to move around our app for it to notice the changes. I added a timer to force a rescan of the DOM tree, then it worked. Of course, that defeats the purpose.

Can't explain it except maybe mutation observers are not as deterministic as we expected. React maybe does on the fly changes to the DOM while the observer is triggered which causes conflicts. This is in Chrome latest version on Linux. Using hooks means we are aligned with the lifecycle of the rendering engine.

So until we resolve this, we'll keep using custom hooks. FYI for anyone watching this issue.
